### PR TITLE
AC-1273 Add exception throwing if file not found

### PR DESCRIPTION
**Pull request by rowellx68**
### Description

Fixes AC-1273

Add exception throwing if file not found

### Status
**Ready**

### Types of changes

- [x] Non-breaking change (fix or new feature that would not break existing functionality).
- [ ] Breaking change (fix or new feature that would cause existing functionality to change).
- [ ] New unit tests added to cover the changes.
- [ ] New e2e tests added to cover the changes. (If it has been decided these will not be added with the PR, a seperate ticket **must** be created and linked in the ticket below before merge)
- [x] All tests passed locally.

### CODE REVIEW

File: src/modules/clinical-review/clinical-review.controller.ts [hunk 1]
```diff
@@ -106,13 +106,13 @@ export class ClinicalReviewController {
...
     }
-    const dicom = await this.service.getDicomThumbnail(roles, key);
+    const thumbnail = await this.service.getDicomThumbnail(roles, key);
 
-    if (!dicom) {
-      res.statusCode = 404;
-    } else {
-      res.send(dicom);
+    if (!thumbnail) {
+      throw new NotFoundException(`Thumbnail not found for '${key}'`);
```
Comment (mehranrashid) [inline src/modules/clinical-review/clinical-review.controller.ts:112]: If the CR service is offline, do you still want it to throw a NotFoundException?

File: src/modules/clinical-review/clinical-review.service.ts [hunk 2]
```diff
@@ -140,13 +140,13 @@ export class ClinicalReviewService {
     );
 
     if (
-      isResultOk(getDicomThumbnail) &&
-      getDicomThumbnail.data instanceof ArrayBuffer &&
-      getDicomThumbnail.data.byteLength !== 0
+      getDicomThumbnail.status === 404 ||
```
Comment (mehranrashid) [inline src/modules/clinical-review/clinical-review.service.ts:143]: Do we want to throw in the service if it's an error status code other than 500?

**Review by answeranand — APPROVED** [review]

**Review by DanielNuttall-AD — APPROVED** [review]

File: src/modules/clinical-review/clinical-review.service.ts (hunk 2 above)
Comment (rowellx68) [reply src/modules/clinical-review/clinical-review.service.ts:143]: I'll add a custom exception that will be handled by the `ExternalExceptionHandler`

Comment (rowellx68) [reply src/modules/clinical-review/clinical-review.service.ts:143]: Actually, I had a look at the code for the `ExternalServerExceptionFilter`, we are actually handling exceptions that might be thrown as a result of requesting data from Clinical Review Service

**Pull request merged**